common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -198,6 +198,7 @@ protected void serviceInit(Configuration conf) throws Exception {
       // register metrics on the block handler into the Node Manager's metrics system.
       blockHandler.getAllMetrics().getMetrics().put("numRegisteredConnections",
           shuffleServer.getRegisteredConnections());
+      blockHandler.getAllMetrics().getMetrics().putAll(shuffleServer.getAllMetrics().getMetrics());
```

Comment:
This one-line seems to be the intended contribution. Did you face some error with `numRegisteredConnections` while adding this? I'm wondering if there is a reason to remove `numRegisteredConnections`.